rdt/hyper_transformer.py

```diff
@@ -73,6 +75,42 @@ class HyperTransformer:
         'b': 'boolean',
         'M': 'datetime',
     }
+    DEFAULT_TRANSFORMERS = {
```

Comment:
I think this could also be moved over to the `transformers` module, and create an additional `get_default_transformer(data_type)` function that has the functionality of looking up the given `data_type` on this dictionary and if not found calls the `get_transformers_by_type` function to choose the first one that it finds that supports it.

Reply:
I don't like the idea of calling `get_transformers_by_type` every time we don't have a defined transformer.

Reply:
What I'll do is add a function that creates the default dict using the `DEFAULT_TRANSFORMERS` and `get_transformers_by_type` dicts, and we can just use that dict in the `HyperTransformer`

Reply:
> What I'll do is add a function that creates the default dict using the `DEFAULT_TRANSFORMERS` and `get_transformers_by_type` dicts, and we can just use that dict in the `HyperTransformer`

I had not seen this. I agree, calling it every time sounds overkill, but somehow I do not like the idea of also calling the function when the HyperTransformer is imported, or every time an instance is created.

What do you think about using [functools.cache](https://docs.python.org/3/library/functools.html#functools.cache)?
We could actually have `get_default_transformers()` which caches its result the first time it is called, but then also have `get_default_transformer(transformer)`, which also caches its results, and that under the hood calls the other one and gets the transformer from the dict.

By doing this, the result is the behavior is the same (we end up having the dictionary stored somewhere instead of building it every time), but its initialization is lazy (the dict gets build on the fly the first time it is used)

Reply:
Makes sense. Let me push a commit and you can take a look